config/v1/types_kmsencryption.go

```diff
@@ -0,0 +1,28 @@
+package v1
+
+// KMSConfig defines the configuration for the KMS instance
+// that will be used with KMSv2 encryption
+//
+// +openshift:enable:FeatureGate=KMSv2
+type KMSConfig struct {
+	// aws defines the key config for using an AWS KMS instance
+	// for the encryption. The AWS KMS instance is managed
+	// by the user outside the purview of the control plane.
+	//
+	// +optional
+	AWS *AWSKMSConfig `json:"aws,omitempty"`
+}
+
+// AWSKMSConfig defines the KMS config specific to AWS KMS provider
+//
+// +openshift:enable:FeatureGate=KMSv2
+type AWSKMSConfig struct {
+	// keyARN is the AWS ARN for the symmetric encryption KMS key
+	//
+	// +kubebuilder:validation:Pattern=`^arn:aws:kms:[a-z0-9-]+:[0-9]{12}:key/[a-f0-9-]+$`
```

Comment:
Use CEL for this, it allows you to craft a custom error message. Break down the rule and explain it to users in concise prose
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.matches('^arn:aws:kms:[a-z0-9-]+:[0-9]{12}:key/[a-f0-9-]+$')",message="keyARN must start with `arn:aws:kms:` followed by ..."
```

Reply:
Added CEL validation.